src/materials/texture.cc

```diff
@@ -53,4 +53,28 @@ color image_texture::value(double u, double v, const point3& p) const {
 
     auto color_scale = 1.0 / 255.0;
     return color(color_scale*pixel[0], color_scale*pixel[1], color_scale*pixel[2]);
+}
+
+// Pixel Image Textures
+pixel_image_texture::pixel_image_texture(const char* filename) : img(filename, 4) {}
+
+color pixel_image_texture::value(double u, double v, const point3& p) const {
+    return color(0,1,1);
+}
+
+RGBA pixel_image_texture::value(double u, double v) const {
+    RGBA result;
```

Comment:
instead of defining it here, and then assigning the fields at the bottom, you could just assign all the values at the bottom and directly return it. up to you though

Reply:
wait what why?